### PR TITLE
Implement interactive mode

### LINKED ISSUE #28: Interactive mode
I'd like to add an interactive mode, which combines `watch` and `profile` without an intermediate file, streaming Cabal output into plotter. Here is a demo (sorry for a large GIF):

render1665770208930

### PR DESCRIPTION
**Pull request by Bodigrim**
Closes #28.

### CODE REVIEW

File: src/DrCabal/Interactive.hs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE NumericUnderscores #-}
+
+{- |
+Module                  : DrCabal.Interactive
+Copyright               : (c) 2019 Alexander Gugel
+                          (c) 2022 Dmitrii Kovanikov
```
Comment (chshersh) [inline src/DrCabal/Interactive.hs:6]: I have no complaints on the current copyright, you can even put your name here if you want 🙂 

But I'm curious about the original code 🤔 Where a similar code was already written?

File: src/DrCabal/Interactive.hs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+watchWorker :: Style -> Int -> Int -> IORef InteractiveState -> IO ()
+watchWorker style terminalWidth chartHeight stateRef = do
+    InteractiveState entries done <- readIORef stateRef
+
+    -- https://github.com/UnkindPartition/ansi-terminal/issues/141
+    when (chartHeight > 0) $
+        cursorUp chartHeight
+    setCursorColumn 0
+    clearFromCursorToScreenEnd
```
Comment (chshersh) [inline src/DrCabal/Interactive.hs:74]: Let's move this into a separate function `clearScreen` or something like that for readability.

**Review by chshersh — COMMENTED** [review]
Actually, this looks pretty good! 👏🏻 

I have only one minor refactoring suggestion but otherwise, I'm glad that multiple parts of the code could be reused 🙂

File: src/DrCabal/Interactive.hs (hunk 1 above)
Comment (Bodigrim) [reply src/DrCabal/Interactive.hs]: My bad, I just copied header from `DrCabal.Watch`.

File: src/DrCabal/Interactive.hs (hunk 2 above)
Comment (Bodigrim) [reply src/DrCabal/Interactive.hs]: Done.

**Review by chshersh — APPROVED** [review]
Thanks! 👏🏻 
That's really great!

**Pull request merged**